### PR TITLE
Bk/fix scroll to item bug 2

### PR DESCRIPTION
**Pull request by bryankeller**
## Details

After I merged this https://github.com/airbnb/HorizonCalendar/pull/17, I found some other bugs and edge cases. I've improved the logic and tested extensively, including dedicated unit tests for these cases.

Ass a reminder, this is fixing the case where you tell the calendar to scroll to the first visible date with a `.centered` scroll position, for example, which is not possible since the first visible date won't be in a natural resting position if it's centered. This logic corrects for that and ensures that our target scroll position is always a valid resting state for the calendar's scroll view.

## Related Issue

N/A

## Motivation and Context

Trying to get this working perfectly!

## How Has This Been Tested

Testing on real device and added unit tests.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: Sources/Internal/VisibleItemsProvider.swift [hunk 1]
```diff
@@ -896,13 +896,14 @@ final class VisibleItemsProvider {
     currentMonth = month
     currentMonthFrame = monthFrame
     while
-      bounds.contains(CGPoint(x: currentMonthFrame.maxX, y: currentMonthFrame.maxY)),
+      bounds.contains(
+        CGPoint(x: currentMonthFrame.maxX - 1, y: currentMonthFrame.maxY - 1)
```
Comment (bryankeller) [inline Sources/Internal/VisibleItemsProvider.swift:900]: `CGRect`'s `containsPoint` function won't return `true` if the point lies on the trailing/bottom edge
